wallet-new/src/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -246,10 +248,14 @@ syncWalletWorker = pure $ const $
 -- | "Attaches" the middleware to this 'Application', if any.
 -- When running in debug mode, chances are we want to at least allow CORS to test the API
 -- with a Swagger editor, locally.
-withMiddleware :: RunMode -> Application -> Application
-withMiddleware wrm app
-  | isDebugMode wrm = corsMiddleware app
-  | otherwise = app
+withMiddleware :: MonadIO m => RunMode -> Application -> m Application
+withMiddleware wrm app = do
+    st <- liftIO $ Throttle.initThrottler
+    pure
+        . (if isDebugMode wrm then corsMiddleware else identity)
+        . Throttle.throttle Throttle.defaultThrottleSettings st
+        $ app
```

Comment:
This is the main change to add throttling. A future PR will bring customized throttling settings from the `configuration.yaml` file.

Reply:
I am curious to see whether this is going to break things like my addition of the HTTP/2.0 middleware breaks things :thinking: ... There's something weird 'about WAI middlewares.